tutorials/jms-output-exactlyonce/src/main/java/com/example/ValidationToFile.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.example;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.apex.malhar.lib.fs.GenericFileOutputOperator;
+
+import com.datatorrent.api.DefaultInputPort;
+
+/**
+ * ValidationToFile operator:
+ *  1. writes all incoming tuples to a validation file on HDFS
+ *  2. when no more tuples come in it checks for duplicates and writes the result to the validation file
+ */
+public class ValidationToFile extends GenericFileOutputOperator.StringFileOutputOperator
+{
+  int lastMessagesSize;
+  int sameMessageSizeCount = 5;
```

Comment:
this is the amount of windows where the size of incoming messages has not changed. After the number of incoming windows has not changed for 5 windows, it is writing the actual validation line. I will add a comment.